src/com/company/exercises/MethodRefs.java

```diff
@@ -0,0 +1,66 @@
+package com.company.exercises;
+
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.function.*;
+
+public class MethodRefs {
+
+    public static void main(String[] args) {
+        // Bound instance method reference (bound to the 's' String instance)
+        String s = "rado";
+        Supplier<String> methodRef = s::toUpperCase;
+        System.out.println(methodRef.get());
+
+        // normal way
+        Consumer<String> greeterFn = greeting -> System.out.println(greeting);
+
+        //method ref way
+        Consumer<String> greeterFn2 = System.out::println;
+        greeterFn2.accept("Hello from a bound instance method reference ");
+
+        //Unbound instance method reference (
+        BiConsumer<PrintStream, String> greeterFn3 = PrintStream::println;
+        //since it's unbound we need to give it instances : System.out for PrintStream and the String literal
+        greeterFn3.accept(System.out,"I'm an unbound instance method reference");
+
+        //Static method reference (currentThread method is static)
+        // currentThread method returns a thread value, so the FI we need is Supplier
+        Supplier<Thread> threadFn = Thread::currentThread;
+        System.out.println("Current thread is: " + threadFn.get());
+
+        // Constructor method reference
+
+        Supplier<Calendar> constrFn = GregorianCalendar::new;
+        Calendar cal = constrFn.get();
+        System.out.println("Calendar : " + cal);
+
+        // Additional examples with lamdba->method ref conversion
+
+        // Static method ref
+
+        Function<String, Double> doubleConverterLamdba = (str) -> Double.parseDouble(str);
```

Comment:
Typo in the word Lambda. Pay attention to the green squiggly line that IntelliJ uses to point you to typos.
Also I would avoid using shortened names like str, txtBox etc.

Reply:
Done

Reply:
doubleConverterLamdba -> doubleConverterLambda

Reply:
Now actually done